internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -919,3 +952,367 @@ func TestMachineSetReconciler_updateStatusResizedCondition(t *testing.T) {
+		g.Eventually(func() bool {
+			if err := env.Get(ctx, client.ObjectKeyFromObject(deletingMachine), deletingMachine); err != nil {
+				return false
+			}
+			return !deletingMachine.DeletionTimestamp.IsZero()
+		}, timeout).Should(BeTrue())
+
+		machines := []*clusterv1.Machine{inPlaceMutatingMachine, deletingMachine}
+
+		// Sync the Machines.
+		reconciler := &Reconciler{Client: env}
+		g.Expect(reconciler.syncMachines(ctx, ms, machines)).To(Succeed())
+
+		// The in-place mutating machine should have:
+		// - clean-up managed fields
+		// - updated in-place propagating values
+		updatedInPlaceMutatingMachine := machines[0]
+		// Verify ManagedFields
+		g.Expect(updatedInPlaceMutatingMachine.ManagedFields).Should(
+			ContainElement(ssa.MatchManagedField(machineSetManagerName, metav1.ManagedFieldsOperationApply)),
+			"in-place mutable machine should contain an entry for SSA manager",
+		)
+		g.Expect(updatedInPlaceMutatingMachine.ManagedFields).ShouldNot(
+			ContainElement(ssa.MatchManagedField("manager", metav1.ManagedFieldsOperationUpdate)),
+			"in-place mutable machine should not contain an entry for old manager",
+		)
+		// Verify Labels
+		g.Expect(updatedInPlaceMutatingMachine.Labels).Should(HaveKeyWithValue("machine-set-matching-label", "true"))
+		// Verify Annotations
+		g.Expect(updatedInPlaceMutatingMachine.Annotations).Should(HaveKeyWithValue("annotation-1", "true"))
+		g.Expect(updatedInPlaceMutatingMachine.Annotations).Should(HaveKeyWithValue("precedence", "MachineSet"))
+		// Verify Node timeout values
+		g.Expect(updatedInPlaceMutatingMachine.Spec.NodeDrainTimeout).Should(And(
+			Not(BeNil()),
+			HaveValue(Equal(*ms.Spec.Template.Spec.NodeDrainTimeout)),
+		))
+		g.Expect(updatedInPlaceMutatingMachine.Spec.NodeDeletionTimeout).Should(And(
+			Not(BeNil()),
+			HaveValue(Equal(*ms.Spec.Template.Spec.NodeDeletionTimeout)),
+		))
+		g.Expect(updatedInPlaceMutatingMachine.Spec.NodeVolumeDetachTimeout).Should(And(
+			Not(BeNil()),
+			HaveValue(Equal(*ms.Spec.Template.Spec.NodeDrainTimeout)),
+		))
+
+		// The deleting machine should not change.
+		updatedDeletingMachine := machines[1]
+		g.Expect(updatedDeletingMachine.ManagedFields).ShouldNot(
+			ContainElement(ssa.MatchManagedField(machineSetManagerName, metav1.ManagedFieldsOperationApply)),
+			"deleting machine should not contain an entry for SSA manager",
+		)
+		g.Expect(updatedDeletingMachine.Labels).Should(Equal(deletingMachine.Labels))
+		g.Expect(updatedDeletingMachine.Annotations).Should(Equal(deletingMachine.Annotations))
+		g.Expect(updatedDeletingMachine.Spec.NodeDrainTimeout).Should(Equal(deletingMachine.Spec.NodeDrainTimeout))
+		g.Expect(updatedDeletingMachine.Spec.NodeDeletionTimeout).Should(Equal(deletingMachine.Spec.NodeDeletionTimeout))
+		g.Expect(updatedDeletingMachine.Spec.NodeVolumeDetachTimeout).Should(Equal(deletingMachine.Spec.NodeVolumeDetachTimeout))
+	})
+}
+
+func TestComputeDesiredMachine(t *testing.T) {
```

Comment:
This test looks fine to me 
(Just because of our previous discussion about changing it)